ui/src/logs/containers/LogsPage.tsx

```diff
@@ -318,12 +323,28 @@ class LogsPage extends Component<Props, State> {
             colorScale={colorForSeverity}
             onZoom={this.handleChartZoom}
             colors={histogramColors}
+            onSortChartBars={this.handleSortHistogramBars}
           />
         )}
       </AutoSizer>
     )
   }
 
+  private handleSortHistogramBars = (): SortFn => {
+    let i = 0
+    const order: {[x: string]: number} = _.reduce(
+      SeverityLevelOptions,
+      (acc, o) => {
+        acc[o] = i
```

Comment:
I think it's a bit confusing to have a non-pure reducer. I think this could be made more explicit by defining a constant like

```
export const SEVERITY_SORTING_ORDER = {
  [SeverityLevelOptions.emerg]: 1,
  [SeverityLevelOptions.alert]: 2,
  [SeverityLevelOptions.crit]: 3,
  [SeverityLevelOptions.err]: 4,
  [SeverityLevelOptions.warning]: 5,
  [SeverityLevelOptions.notice]: 6,
  [SeverityLevelOptions.info]: 7,
  [SeverityLevelOptions.debug]: 8,
}
```

and then the sorting function would be
```
(a, b) => SEVERITY_SORTING_ORDER[b.group] - SEVERITY_SORTING_ORDER[a.group]
```

Reply:
totally, i will make that change.